modules/document_repository/jsx/categoryForm.js

```diff
@@ -0,0 +1,211 @@
+      uploadProgress: -1,
+    };
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.isValidForm = this.isValidForm.bind(this);
+    this.setFormData = this.setFormData.bind(this);
+    this.uploadFile = this.uploadFile.bind(this);
+    this.fatchData = this.fetchData.bind(this);
+  }
+ fetchData() {
+    return fetch(this.props.DataURL, {credentials: 'same-origin'})
+      .then((resp) => resp.json())
+      .then((data) => this.setState({data: data, isLoaded: true}))
+      .catch((error) => {
+        this.setState({error: 'An error occurred when loading the form!'});
+        console.error(error);
+      });
+}
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  render() {
+    // Data loading error
+    if (this.state.error !== undefined) {
+      return (
+        <div className="alert alert-danger text-center">
+          <strong>
+            {this.state.error}
+          </strong>
+        </div>
+      );
+    }
+
+    // Waiting for data to load
+    if (!this.state.isLoaded) {
+      return (
+        <button className="btn-info has-spinner">
+          Loading
+          <span
+            className="glyphicon glyphicon-refresh glyphicon-refresh-animate">
+          </span>
+        </button>
+      );
+    }
+
+    return (
+      <div className="row">
+        <div className="col-md-8 col-lg-7">
+          <FormElement
+            name="docUpload"
+            fileUpload={true}
+            onSubmit={this.handleSubmit}
+            ref="form"
+          >
+            <h3>Add a category</h3><br/>
+            <TextboxElement
+              name="category_name"
+              label="Category Name"
+              onUserInput={this.setFormData}
+              ref="category_name"
+              required={true}
+              value={this.state.formData.category_name}
+            />
+            <SelectElement
+              name="parent_id"
+              label="Parent"
+              options={this.state.data.fieldOptions.fileCategories}
+              onUserInput={this.setFormData}
+              ref="parent_id"
+              hasError={false}
+              value={this.state.formData.parent_id}
+            />
+            <TextareaElement
+              name="comments"
+              label="Comments"
+              onUserInput={this.setFormData}
+              ref="comments"
+              value={this.state.formData.comments}
+            />
+            <ButtonElement label="Add Category"/>
+          </FormElement>
+        </div>
+      </div>
+    );
+  }
+
+/** *******************************************************************************
+ *                      ******     Helper methods     *******
+ *********************************************************************************/
+
+
+  /**
+   * Handle form submission
+   * @param {object} e - Form submission event
+   */
+  handleSubmit(e) {
+    e.preventDefault();
+
+    let formData = this.state.formData;
+    let formRefs = this.refs;
```

Comment:
Refs should be used here. Error messages should be tracked via a error state object.